sdk/storage/azure-storage-file-datalake/src/test/java/com/azure/storage/file/datalake/FileAsyncApiTests.java

```diff
@@ -424,12 +433,3792 @@ public void createOptionsWithTimeToExpireRelativeToNow() {
+            Arguments.of(3L, 2L, DATA.getDefaultText().substring(3, 3 + 2))
+        );
+    }
+
+    @ParameterizedTest
+    @MethodSource("modifiedMatchAndLeaseIdSupplier")
+    public void readAC(OffsetDateTime modified, OffsetDateTime unmodified, String match, String noneMatch,
+                       String leaseID) {
+        DataLakeRequestConditions drc = new DataLakeRequestConditions()
+            .setLeaseId(setupPathLeaseCondition(fc, leaseID))
+            .setIfMatch(setupPathMatchCondition(fc, match))
+            .setIfNoneMatch(noneMatch)
+            .setIfModifiedSince(modified)
+            .setIfUnmodifiedSince(unmodified);
+
+        StepVerifier.create(fc.readWithResponse(null, null, drc, false))
+            .assertNext(r -> assertEquals(200, r.getStatusCode()))
+            .verifyComplete();
+    }
+
+    @ParameterizedTest
+    @MethodSource("invalidModifiedMatchAndLeaseIdSupplier")
+    public void readACFail(OffsetDateTime modified, OffsetDateTime unmodified, String match, String noneMatch,
+                           String leaseID) {
+        setupPathLeaseCondition(fc, leaseID);
+        DataLakeRequestConditions drc = new DataLakeRequestConditions()
+            .setLeaseId(leaseID)
+            .setIfMatch(match)
+            .setIfNoneMatch(setupPathMatchCondition(fc, noneMatch))
+            .setIfModifiedSince(modified)
+            .setIfUnmodifiedSince(unmodified);
+
+
+        StepVerifier.create(fc.readWithResponse(null, null, drc, false))
+                .verifyError(DataLakeStorageException.class);
+    }
+
+    @Test
+    public void readMd5() throws NoSuchAlgorithmException {
+        fc.append(DATA.getDefaultBinaryData(), 0).block();
+        fc.flush(DATA.getDefaultDataSizeLong(), true).block();
+
+        StepVerifier.create(fc.readWithResponse(new FileRange(0, 3L),
+                null, null, true))
+            .assertNext(r -> {
+                byte[] contentMD5 = r.getHeaders().getValue(HttpHeaderName.CONTENT_MD5).getBytes();
+                try {
+                    TestUtils.assertArraysEqual(
+                        Base64.getEncoder().encode(
+                            MessageDigest.getInstance("MD5").digest(DATA.getDefaultText().substring(0, 3).getBytes())),
+                        contentMD5);
+                } catch (NoSuchAlgorithmException e) {
+                    throw new RuntimeException(e);
+                }
+            })
+            .verifyComplete();
+    }
+
+    @Test
+    public void readRetryDefault() {
+        fc.append(DATA.getDefaultBinaryData(), 0).block();
+        fc.flush(DATA.getDefaultDataSizeLong(), true).block();
+        DataLakeFileAsyncClient failureFileAsyncClient = getFileAsyncClient(getDataLakeCredential(), fc.getFileUrl(),
+            new MockFailureResponsePolicy(5));
+
+        ByteArrayOutputStream downloadData = new ByteArrayOutputStream();
+
+        StepVerifier.create(failureFileAsyncClient.read())
```

Comment:
I might recommend using `FluxUtil.collectBytesInByteBufferStream` to convert the `read` to a `byte[]` and then that can be compared to `DATA.getDefaultBytes()`

```suggestion
        StepVerifier.create(FluxUtil.collectBytesInByteBufferStream(failureFileAsyncClient.read()))
            .assertNext(bytes -> TestUtils.assertArraysEqual(DATA.getDefaultBytes(), bytes))
            .verifyComplete();
```